src/org/labkey/test/tests/RenameFolderJavaClientApiTest.java

```diff
@@ -0,0 +1,153 @@
+import java.util.List;
+
+@Category({Daily.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 4)
+public class RenameFolderJavaClientApiTest extends BaseWebDriverTest
+{
+    private static final String DUPLICATE_FOLDER_NAME = "Duplicate folder name";
+
+    @Override
+    protected @Nullable String getProjectName()
+    {
+        return getClass().getSimpleName() + " Project";
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return null;
+    }
+
+    @BeforeClass
+    public static void doSetup()
+    {
+        RenameFolderJavaClientApiTest initTest = (RenameFolderJavaClientApiTest) getCurrentTest();
+        initTest.setupProject();
+    }
+
+    @LogMethod
+    private void setupProject()
+    {
+        _containerHelper.createProject(getProjectName());
+        _containerHelper.createProject(DUPLICATE_FOLDER_NAME);
+    }
+
+    @Test
+    public void testRenameFolder() throws IOException, CommandException
+    {
+        String newContainerName = "Renamed Name for " + getProjectName();
+        String newTitle = "Renamed Title for " + getProjectName();
+
+        goToProjectHome();
+        Connection cn = WebTestHelper.getRemoteApiConnection();
+
+        log("New name only: Verify name and title become " + newContainerName);
+        RenameContainerCommand command = new RenameContainerCommand(newContainerName, null, true);
+        RenameContainerResponse response = command.execute(cn, getProjectName());
+        Assert.assertEquals("Rename container api failed ", 200, response.getStatusCode());
+        Assert.assertEquals("Incorrect new name", newContainerName, response.getName());
+        Assert.assertEquals("Incorrect new title", newContainerName, response.getProperty("title"));
+
+        log("New title only: Verify only title changes to " + newTitle);
+        command = new RenameContainerCommand(null, newTitle, true);
+        response = command.execute(cn, getProjectName());
+        Assert.assertEquals("Rename container api failed ", 200, response.getStatusCode());
+        Assert.assertEquals("Incorrect new name", newContainerName, response.getName());
+        Assert.assertEquals("Incorrect new title", newTitle, response.getProperty("title"));
+
+        log("New name and title: Verify both name and title update");
+        command = new RenameContainerCommand(newContainerName + "1", newTitle + "1", true);
+        response = command.execute(cn, getProjectName());
+        Assert.assertEquals("Rename container api failed ", 200, response.getStatusCode());
+        Assert.assertEquals("Incorrect new name", newContainerName + "1", response.getName());
+        Assert.assertEquals("Incorrect new title", newTitle + "1", response.getProperty("title"));
+
+        log("Required values missing: Verify error appears");
+        command = new RenameContainerCommand(null, null, false);
+        try
+        {
+            command.execute(cn, getProjectName());
+        }
```

Comment:
If you're expecting an error, this should fail when an error doesn't occur.
```suggestion
        try
        {
            command.execute(cn, getProjectName());
            Assert.fail("Rename should have failed");
        }
```